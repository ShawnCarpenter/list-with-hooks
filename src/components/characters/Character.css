.Character {
  a {
    text-decoration: none;
    color: #272727;
    display: flex;
    flex-direction: column;
  }
  img {
    max-height: 75px;
    max-width: 75px;
    padding: .2rem;
    border-radius: 1rem;
  }
}
