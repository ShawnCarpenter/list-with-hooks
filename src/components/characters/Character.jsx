import React from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';
const Character = ({ name, image }) => {
  return (
    <div className={styles.Character}>
      <a href={`detail/${name}`}>
        <h3>{name}</h3>
        <img src={image} alt={name} />
      </a>
    </div>
    
  );
};

Character.propTypes = {
  name:PropTypes.string.isRequired,
  image:PropTypes.string.isRequired
};

export default Character;
