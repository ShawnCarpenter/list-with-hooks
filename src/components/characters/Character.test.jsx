import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';


describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(
      <Character name="Bob" image="http://image.com/pic.jpg"
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
