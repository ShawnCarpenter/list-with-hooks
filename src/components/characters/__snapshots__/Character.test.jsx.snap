// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <div
    class="Character"
  >
    <a
      href="detail/Bob"
    >
      <h3>
        Bob
      </h3>
      <img
        alt="Bob"
        src="http://image.com/pic.jpg"
      />
    </a>
  </div>
</DocumentFragment>
`;
